### PR TITLE
feat: Upgrade edx-event-bus-kafka to 3.9.1 (adds audit logging)

### PR DESCRIPTION
**Pull request by timmc-edx**
openedx-events major version bump is a change to how consumers send
metadata to signal receivers, and is coordinated with the event-bus-kafka
3.8.1 version.

### CODE REVIEW

File: requirements/local.txt [hunk 1]
```diff
@@ -521,7 +521,7 @@ oauthlib==3.2.2
     # via
     #   requests-oauthlib
     #   social-auth-core
-openedx-events==4.2.0
+openedx-events==5.0.0
```
Comment (robrap) [inline requirements/local.txt:524]: Is this major upgrade related? Can you update the PR description to clarify what is going on here? Thanks.

Comment (timmc-edx) [reply requirements/local.txt:524]: It's related -- it's the breaking change relating to event metadata that's coordinated with the event-bus-kafka 3.8.1 change.

Comment (timmc-edx) [reply requirements/local.txt:524]: Rewrote commit and updated PR desc.

**Review by robrap — APPROVED** [review]
Thank you!

**Comment by timmc-edx** [comment]
Upgrade has since happened on another PR.
